### PR TITLE
Setup OpenTelemetry exporters when initializing prefect

### PR DESCRIPTION
**Pull request by bunchesofdonald**
This PR adds OpenTelemetry support to Prefect Cloud by setting up exporters for traces, metrics, and logs during initialization when enabled.

Key changes:
- Adds experimental setting to enable/disable telemetry
- Add OTLP exporters for traces, metrics, and logs
- Implement custom InFlightSpanProcessor for handling long-running spans
- Configure telemetry endpoint URL and auth based on existing Prefect settings

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -36,3 +36,12 @@ mkdocs
 mkdocs-material
 mkdocstrings[python]
 mkdocs-gen-files
+
+# open-telemetry
```
Comment (chrisguidry) [inline requirements-dev.txt:40]:
```suggestion
# OpenTelemetry
```
is usually how it's styled

**Review by chrisguidry — COMMENTED** [review]
Great tests!  I'm thinking we need to move some of the imports conditionally into functions?

**Comment by bunchesofdonald** [comment]
> Great tests! I'm thinking we need to move some of the imports conditionally into functions?

Yeah, I'm working on something like that, having a `bootstrap.py` that handles knowing if the imports should happen.

**Review by desertaxle — APPROVED** [review]

File: src/prefect/telemetry/bootstrap.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+):
+    settings = prefect.settings.get_current_settings()
+    if not settings.experiments.telemetry_enabled:
+        return None, None, None
+
+    server_type = determine_server_type()
+    if server_type != ServerType.CLOUD:
+        return None, None, None
+
+    assert settings.api.key
```
[the remark this answers is not in the document]
Comment (bunchesofdonald) [reply src/prefect/telemetry/bootstrap.py:26]: Yeah, I have a ticket to come through and make this a more friendly thing in general: https://linear.app/prefect/issue/CLOUD-675/make-telemetry-failures-more-friendly

**Pull request merged**